packages/esm-patient-chart-app/src/visit/past-visit-overview.component.tsx

```diff
@@ -0,0 +1,202 @@
+import { Visit } from '@openmrs/esm-api';
+import React, { useState, useEffect, useMemo, useReducer, useCallback } from 'react';
+import DataTable, {
+  TableContainer,
+  Table,
+  TableBody,
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableCell,
+  DataTableHeader,
+  DataTableRow,
+} from 'carbon-components-react/es/components/DataTable';
+import DataTableSkeleton from 'carbon-components-react/es/components/DataTableSkeleton';
+import OverflowMenu from 'carbon-components-react/es/components/OverflowMenu';
+import OverflowMenuItem from 'carbon-components-react/es/components/OverflowMenuItem';
+import { useTranslation } from 'react-i18next';
+import dayjs from 'dayjs';
+import { ErrorState } from '@openmrs/esm-patient-common-lib';
+import { attach, detach, getStartedVisit, getVisitsForPatient, VisitMode, VisitStatus } from '@openmrs/esm-framework';
+import styles from './past-visit-overview.component.scss';
+import Button from 'carbon-components-react/es/components/Button';
+
+enum ActionTypes {
+  pending = 'pending',
+  resolved = 'resolved',
+  error = 'error',
+}
+interface Pending {
+  type: ActionTypes.pending;
+}
+
+interface Error {
+  type: ActionTypes.error;
+  payload: Error;
+}
+
+interface Resolved {
+  type: ActionTypes.resolved;
+  payload: Array<Visit>;
+}
+
+type Action = Pending | Error | Resolved;
+
+interface PastVisitOverviewState {
+  status: 'pending' | 'resolved' | 'error';
+  patientPastVisits: Array<Visit>;
+  error?: null | Error;
+}
+
+function reducer(state: PastVisitOverviewState, action: Action): PastVisitOverviewState {
+  switch (action.type) {
+    case ActionTypes.pending:
+      return {
+        status: 'pending',
+        ...state,
+      };
+    case ActionTypes.resolved:
+      return {
+        status: 'resolved',
+        patientPastVisits: action.payload,
+        error: null,
+      };
+    case ActionTypes.error:
+      return {
+        status: 'error',
+        patientPastVisits: null,
+        error: action.payload,
+      };
+    default:
+      return state;
+  }
+}
+
+interface PastVisitOverviewProps {
+  patientUuid: string;
+}
+
+const PastVisitOverview: React.FC<PastVisitOverviewProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const [{ status, patientPastVisits, error }, dispatch] = useReducer(reducer, {
+    status: 'pending',
+    patientPastVisits: [],
```

Comment:
```suggestion
    patientPastVisits: null,
```

Or else make it set to `[]` in the "error" action of the reducer, instead of `null`.